5.process_features/scripts/1.preprocess_features.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import umap
+from pycytominer import feature_select, normalize
+from sklearn.decomposition import PCA
+
+# In[2]:
+
+
+df_path = pathlib.Path("../../data/5.converted_data/output.parquet").resolve(
+    strict=True
+)
+
+df = pd.read_parquet(df_path)
+
+
+# In[3]:
+
+
+df["Metadata_genotype"] = df["Metadata_Image_FileName_OP"].str.split("_").str[1]
+df["Metadata_replicate"] = df["Metadata_Image_FileName_OP"].str.split("_").str[2]
+df["Metadata_side"] = (
+    df["Metadata_Image_FileName_OP"].str.split("_").str[3].str.split(".tiff").str[0]
+)
+df["Metadata_genotype"].value_counts()
+
+
+# In[4]:
+
+
+# get the columns that contain nan
+nan_columns = df.columns[df.isna().any()].tolist()
+# pop AreaShape_FormFactor from nan_columns list
+nan_columns.pop(0)
+nan_columns
+# remove nan from the dataframe
+df = df.drop(nan_columns, axis=1)
+df.head(2)
+
+
+# In[5]:
+
+
+# split the data into metadata and features by columns that contain metadata
+metadata = df.columns[df.columns.str.contains("Metadata")]
+metadata_df = df[metadata]
+features_df = df.drop(metadata, axis=1)
+
+print(df.shape, metadata_df.shape, features_df.shape)
```

Comment:
What is the goal of separating the metadata from features? It looks like you also print the shapes in lines 76 and 83 so maybe including code comment on why this is needed would be sufficient.